src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UICaseSearchController.java

```diff
@@ -0,0 +1,124 @@
+        value = "Search cases according to the provided ElasticSearch query. Supports searching across multiple case types and a use case.",
+        notes = V2.EXPERIMENTAL_WARNING
+    )
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
+            message = "Success.",
+            response = CaseSearchResultViewResource.class
+        ),
+        @ApiResponse(
+            code = 400,
+            message = "Request is invalid. For some other types HTTP code 500 is returned instead.\n"
+                      + "Examples include:\n"
+                      + "- Unsupported use case specified in `usecase` query parameter.\n"
+                      + "- Query is missing required `query` field.\n"
+                      + "- Query includes blacklisted type.\n"
+                      + "- Query has failed in ElasticSearch - for example, a sort is attempted on an unknown/unmapped field."
+        ),
+        @ApiResponse(
+            code = 401,
+            message = "Request doesn't include a valid `Authorization` header. "
+                      + "This applies to all missing, malformed & expired tokens."
+        ),
+        @ApiResponse(
+            code = 403,
+            message = "Request doesn't include a valid `ServiceAuthorization` header. "
+                      + "This applies to all missing, malformed & expired tokens.\n\n"
+                      + "A valid S2S token issued to the name of a non-permitted API Client will also return the same."
+        ),
+        @ApiResponse(
+            code = 404,
+            message = "Case type specified in `ctid` query parameter could not be found."
+        ),
+        @ApiResponse(
+            code = 500,
+            message = "An unexpected situation that is not attributable to the user or API Client; or request is invalid. "
+                      + "For some other types HTTP code 400 is returned instead.\n"
+                      + "Invalid request examples include:\n"
+                      + "- Malformed JSON request."
+        )
+    })
+    @LogAudit(operationType = AuditOperationType.SEARCH_CASE, caseTypeIds = "#caseTypeIds",
+        caseId = "T(uk.gov.hmcts.ccd.v2.internal.controller.UICaseSearchController).buildCaseIds(#result)")
+    public ResponseEntity<CaseSearchResultViewResource> searchCases(
+                                     @ApiParam(value = "Comma-separated list of case type ID(s).")
+                                     @RequestParam(value = "ctid", required = false) List<String> caseTypeIds,
+                                     @ApiParam(value = "Use case for search. Examples include `WORKBASKET`, `SEARCH` or `ORGCASES`.")
+                                     @RequestParam(value = "usecase", required = false) final String useCase,
+                                     @ApiParam(value = "Native ElasticSearch Search API request. Please refer to the ElasticSearch official "
+                                         + "documentation.",
```

Comment:
can you also point them to the elastic search API LLD please: https://tools.hmcts.net/confluence/display/RCCD/CCD+ElasticSearch+and+new+search+API+Design